### PR TITLE
choose min players in game setup

### PR DESCRIPTION
**Pull request by marndt93**
resolves: #6538

### CODE REVIEW

File: migrate/0001_initial.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@
       String :description, null: false
 
       String :title, null: false
+      Integer :min_players, null: false, default: 3
```
Comment (tobymao) [inline migrate/0001_initial.rb:36]: you can't mutate a migration like this, you have to add a brand new one

File: models/game.rb [hunk 2]
```diff
@@ -127,6 +127,7 @@ def to_h(include_actions: false, player: nil)
       description: description,
       user: user.to_h,
       players: ordered_players.map(&:to_h),
+      min_players: defined?(min_players) ? min_players : nil,
```
Comment (tobymao) [inline models/game.rb:130]: what's going on here this looks bad

File: models/game.rb [hunk 3]
```diff
@@ -127,6 +127,7 @@ def to_h(include_actions: false, player: nil)
       description: description,
       user: user.to_h,
       players: ordered_players.map(&:to_h),
+      min_players: min_players || nil,
```
Comment (tobymao) [inline models/game.rb:130]: why do you need an || nil

File: migrate/0006_add_min_players_game_setting.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    add_column :games, :min_players, Integer, null: false, default: 3
```
Comment (tobymao) [inline migrate/0006_add_min_players_game_setting.rb:5]: would it make more sense to move max_players and min_players into settings?

File: models/game.rb (hunk 3 above)
Comment (marndt93) [reply models/game.rb:130]: I am not sure if it is necessary, but I was worried about games created before this property was added.

Comment (tobymao) [reply models/game.rb:130]: no it's not necessary

File: migrate/0006_add_min_players_game_setting.rb (hunk 4 above)
Comment (marndt93) [reply migrate/0006_add_min_players_game_setting.rb:5]: possibly.  Would you like this change made?

You are talking about the settings json obj that is already in the games db table?

Comment (tobymao) [reply migrate/0006_add_min_players_game_setting.rb:5]: right, in the settings, the migration will be a bit more involved

**Comment by tobymao** [comment]
i'm expecting a migration of max_players

**Comment by marndt93** [comment]
> i'm expecting a migration of max_players

max_players column should be removed, right?

**Comment by tobymao** [comment]
yes

**Comment by tobymao** [comment]
you need to iterate every game and move the deleted settings into the json, please test this out on games in your db, if it’s too involved, i’d rather just close this pr

**Comment by marndt93** [comment]
> you need to iterate every game and move the deleted settings into the json, please test this out on games in your db, if it’s too involved, i’d rather just close this pr

hmm okay.  I can do this, but can you tell me where the local db is, so I can view the contents to verify?

If you want, we can close this PR for now and I can re-open it when I am finished with the migration

**Comment by tobymao** [comment]
just make a couple of games on your local server

**Comment by tobymao** [comment]
what do you mean view local?

**Comment by marndt93** [comment]
> what do you mean view local?

I need to make some games with the old db configuration, then write and run a migration that moves the max_players to the settings and then drops the max_players column.

Wouldn't it be ideal to open the database and view the tables to verify that the data was moved successfully?  The docker script creates a postgres db locally, doesn't it?

**Comment by tobymao** [comment]
look at the docker compose for the port, you can use any postgres client to look at the local db

**Comment by vandamm** [comment]
@marndt93 something like this should be sufficient :
```ruby
Sequel.migration do
  transaction
  up do
    run 'update games set settings = jsonb_set(settings, '{max_players}', to_jsonb(max_players))'

    alter_table :games do
      drop_column :max_players
    end
  end

  down do
    alter_table :games do
      add_column :max_players, Integer, null: false, default: 6
    end  

    run 'update games set max_players = (settings->>'max_players' || 6)::integer;'
  end
end
```

**Comment by marndt93** [comment]
@vandamm does this drop the max_players column and then add it back again?

**Comment by vandamm** [comment]
@marndt93 it drops in direct migration, but adds back if you want to rollback

**Comment by marndt93** [comment]
@vandamm I was just testing this today, but I am getting some errors.  I am not too experienced with migrations, so not too sure what the issue is.  Should we be using jsonb_insert instead of jsonb_set?

This is the error:
`rack_1   | E, [2022-03-13T18:33:29.918603 #8] ERROR -- : PG::UndefinedColumn: ERROR:  column "{max_players}" does not exist
rack_1   | LINE 1: ...pdate games set settings = jsonb_insert(settings, "{max_play...
rack_1   |                                                              ^
rack_1   | HINT:  Perhaps you meant to reference the column "games.max_players".: update games set settings = jsonb_insert(settings, "{max_players}", to_jsonb(max_players))`

**Pull request closed without merging**